coursier.bzl

```diff
@@ -497,6 +499,17 @@ def _pinned_coursier_fetch_impl(repository_ctx):
 
     http_files.extend(_get_jq_http_files())
 
+    http_files.extend([
+        "maven_artifacts = [%s]" % (",".join([
```

Comment:
This structure doesn't match the expected usage. If you tried to add these `maven_artifacts` to a target's `runtime_deps`, the build will fail, as the deps are expected to be labels and not structs.

I think there's a couple of ways of handling this:

1. Document this clearly in the README.md and explain how people could use it.
2. Use the maven coordinates directly, once they've been converted back from the dict they are to a string, and use the `artifact` macro to ensure they're loaded properly.

The first option is a lot less work, but I think the second one makes the output more usable.

Reply:
one slight complication with coords is that `artifact` macro only works with `"groupId:artifactId"` and that there's strip classifier logic, need some work to extend `artifact` macro

for example for `com.github.jnr:jffi:jar:native:1.3.1`, `com.github.jnr:jffi:jar:sources:1.3.1` and `com.github.jnr:jffi:1.3.1` there are following targets
```
@maven//:com_github_jnr_jffi_native_1_3_1 // alias
@maven//:com_github_jnr_jffi_native // actual import
@maven//:com_github_jnr_jffi_jar_sources_1_3_1_extension
@maven//:com_github_jnr_jffi_jar_native_1_3_1_extension
@maven//:com_github_jnr_jffi_1_3_1_extension
@maven//:com_github_jnr_jffi_1_3_1 // alias
@maven//:com_github_jnr_jffi // import
```
so `artifact("com.github.jnr:jffi:jar:native:1.3.1")` needs to also strip `jar` classifier it seems

Reply:
The group and artifact IDs are the first two fields in the maven coordinates, which makes life a bit easier. I think you'd need to gather together all the imports, transform, and then sort, removing duplicates, and you'd be okay.

Reply:
Currenty I can put `com.github.jnr:jffi:jar:native:1.3.1` and `com.github.jnr:jffi:1.3.1` as artifacts in `maven_install` and then refer to them as `@maven//:com_github_jnr_jffi_native` and `@maven//:com_github_jnr_jffi`.

But there seems to be no ready-made macro yet that'd transform string `"com.github.jnr:jffi:jar:native:1.3.1"` to label `@maven//:com_github_jnr_jffi_native`

```python
load("@rules_jvm_external//:defs.bzl", "artifact")
load("@rules_jvm_external//:specs.bzl", "parse")

s = "com.github.jnr:jffi:jar:native:1.3.1"
artifact(s) ==> @maven//:com_github_jnr_jffi # without _native part
parse.parse_maven_coordinate(s) ==> { group: "com.github.jnr", artifact: "jffi", classifier: "native", packaging: "jar", version: "1.3.1" }
artifact(parse.parse_maven_coordinate(s)) ==> still @maven//:com_github_jnr_jffi
```
I could decypher and replicate dict -> label conversion (aka `escape(strip_packaging_and_classifier_and_version(s))`), but maybe instead `artifact` macro should be extended to handle all potential keys (it already can accept dict, but decides to only use group and artifact)

Reply:
Sounds quite doable, but maybe there's an even better way?

Reply:
Update PR with support for full string coords in `artifact` macro